pkg/minikube/cluster/cluster.go

```diff
@@ -587,3 +699,30 @@ func EnsureMinikubeRunningOrExit(api libmachine.API) {
 		os.Exit(1)
 	}
 }
+
+func addMinikubeAddonsDirToAssets(assetList *[]CopyableFile) error {
+	// loop over .minikube/addons and add them to assets
+	searchDir := constants.MakeMiniPath("addons")
+	fileList := []string{}
+	err := filepath.Walk(searchDir, func(path string, f os.FileInfo, err error) error {
+		fileList = append(fileList, path)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	for _, addonFile := range fileList {
```

Comment:
Could you do this inside the walkFn above instead of storing a separate list, then iterating over it?